engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -3285,7 +3285,16 @@ namespace dmGraphics
             if (texture->m_Format != vk_format || texture->m_Width != params.m_Width || texture->m_Height != params.m_Height)
             {
                 DestroyResourceDeferred(g_VulkanContext->m_MainResourcesToDestroy[g_VulkanContext->m_SwapChain->m_ImageIndex], texture);
-                texture->m_Format = vk_format;
+                texture->m_Format      = vk_format;
+                texture->m_Width       = params.m_Width;
+                texture->m_Height      = params.m_Height;
+
+                // If the texture has requested mipmaps and we need to recreate the texture, make sure to allocate enough mipmaps
+                // JG: Might want to go back to this part and recreate if mipmap count has changed from when the texture was originally created
+                if (texture->m_MipMapCount > 1)
+                {
+                    texture->m_MipMapCount = (uint16_t) GetMipmapCount(dmMath::Max(texture->m_Width, texture->m_Height));
+                }
```

Comment:
This change is here to reflect the same behaviour as OpenGL - if the texture was once created with mipmaps and we need to resize the texture we have to update the size of it as well!